src/realm/object-store/results.cpp

```diff
@@ -935,6 +935,12 @@ Query Results::get_query() const
     return do_get_query();
 }
 
+DescriptorOrdering Results::get_ordering() const REQUIRES(!m_mutex)
```

Comment:
You can just return `const DescriptorOrdering&` here.
There should be no reason to lock the mutex. m_descriptor_ordering is not guarded by that.

Reply:
I am not sure, it seems to me that both:
`Results Results::sort(SortDescriptor&& sort)` and `Results Results::limit(size_t max_count) const` grab the lock before to alter the ordering...

Reply:
I think those functions acquire the lock slightly earlier than they need to. The mutex is there only for frozen Results and needs to guard the fields which are mutated on _read_, such as `m_collection`, `m_table_view` and `m_query`. `m_descriptor_ordering` is assigned on construction and then is never mutated, so unlocked access from multiple threads is safe.